Sources/_StringProcessing/RegexDSL/DSLConsumers.swift

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension MatchingCollectionConsumer where Consumed == String {
+  public var regex: Regex<Match> {
+    Regex(node: .matcher(.init(Match.self)) {
+      self.matchingConsuming($0, in: $1)
+    })
+  }
+}
+
+extension CollectionConsumer where Consumed == String {
+  public var regex: Regex<Void> {
+    Regex(node: .consumer {
+      self.consuming($0, in: $1)
+    })
+  }
+}
```

Comment:
I don't think we'll need these once we add the `CustomRegexComponent` protocol (but I can remove these when that happens).

Reply:
Before we have that protocol, do we also want overloads that take `Substring`?